public/scripts/ssooverride.js

```diff
@@ -83,33 +89,53 @@
         }, false);
       });
 
-      var thimbleAuth = new WebmakerLogin({
-        csrfToken: csrf,
-        showCTA: false
-      });
+      var basicQuery = "";
+
+      basicQuery += "?client_id=" + oauthClientId;
+      basicQuery += "&response_type=code";
+      basicQuery += "&scopes=user email";
 
+      // Signup login flow
       joinEl.on('click', function() {
-        thimbleAuth.create();
+        var state = uuid.v4();
+        var oauthRoute = "/login/oauth/authorize";
+
+        var query = authHostname + oauthRoute + basicQuery;
+        query += "&state=" + state;
+        query += "&action=signup";
+
+        setStateCookie(state);
+
+        window.location = query;
       });
+      // Login flow
       loginEl.on('click', function() {
-        thimbleAuth.login();
+        var state = uuid.v4();
+        var oauthRoute = "/login/oauth/authorize";
+
+        var query = authHostname + oauthRoute + basicQuery;
+        query += "&state=" + state;
+        query += "&action=signin";
+
+        setStateCookie(state);
+
+        window.location = query;
       });
       logoutEl.on('click', function() {
-        thimbleAuth.logout();
-      });
+        // Logout flow
+        var oauthRoute = "/logout";
+
+        var query = authHostname + oauthRoute;
+        query += "?client_id=" + oauthClientId;
 
-      // Attach event listeners!
-      thimbleAuth.on('login', enable);
-      thimbleAuth.on('logout', disable);
-      thimbleAuth.on('verified', function(user) {
-        if (user) {
-          enable(user);
-        } else {
-          disable();
-        }
+        window.location = query;
       });
 
-      // Default state is signed-out
+      if (user.username.length > 0) {
```

Comment:
I approve of return often, return early, but this is overly prudent, since the function doesn't return anything. In this case, a normal if/else makes more sense
```
if (user.username.length > 0) { enable(user); } else { disable(); }
```
```